mesonbuild/build.py

```diff
@@ -1704,6 +1704,9 @@ class FileInTargetPrivateDir:
     def __init__(self, fname: str):
         self.fname = fname
 
+    def __repr__(self) -> str:
+        return self.fname
```

Comment:
Seems wrong to misuse repr for this. Didn't we use File in all cases where FileInTargetPrivateDir now gets used? What was the exact display before?

Reply:
sole_output is guaranteed to be a string at this stage, even if that string is the string:

```
sole_output = str('<mesonbuild.build.FileMaybeInTargetPrivateDir object at 0x000001C47609D5D0>')
```

Reply:
I see... Therefore this is `__str__` that is needed for `FileInTargetPrivateDir`.

Reply:
@eli-schwartz Is it good for you?